File: ios/AuthClient/Models/ApiClientResult.swift
///ApiClientResult.swift
///AuthClient-iOS
///Created by Rahul Narayanan Unni on 28/07/24..
///Lead Software Engineer @Ospyn Technologies Limited



import Foundation


struct ApiClientResult:Codable {
  //private JsonElement data;
  public let error:Bool?;
  public let errorMessage:String?;
  public let errorReason:Int?;
  public let encryptedContent:String?;
  public let success:Bool?;
  public let message:String?;
  public let errorCode:Int?;
  let data: ErrorData?

}


struct ErrorData: Codable {
    let isError: Bool
    let errorMessage: String
}

File: android/src/main/java/com/reactnativeauthclient/models/ApiClientResult.kt
package com.reactnativeauthclient.models

import com.google.gson.JsonElement
import com.google.gson.annotations.SerializedName

data class ApiClientResult(
    @SerializedName("error")
    var error: Boolean = false,
    
    @SerializedName("errorMessage")
    var errorMessage: String? = null,
    
    @SerializedName("errorReason")
    var errorReason: Int = 0,
    
    @SerializedName("apiResponse")
    var apiResponse: JsonElement? = null,
    
    @SerializedName("data")
    var data: JsonElement? = null,
    
    @SerializedName("success")
    var success: Boolean = false,
    
    @SerializedName("message")
    var message: String? = null,
    
    @SerializedName("errorCode")
    var errorCode: String? = null
)